autoload/neomake/makers/ft/haskell.vim

```diff
@@ -59,7 +65,36 @@ function! neomake#makers#ft#haskell#hdevtools() abort
             \ '%+C  %#%m,'.
             \ '%W%>%f:%l:%v:,'.
             \ '%+C  %#%tarning: %m,'
-        \ })
+        \ }
+    " hdevtools needs the GHC-PACKAGE-PATH environment variable to exist
+    " when running on a project WITHOUT a cabal file, but it needs the
+    " GHC-PACKAGE-PATH to NOT exist when running on a with a project WITH
+    " a cabal file
+    let s:HdevPP = 1
+    if executable('stack')
+        let command =
+              \   'ls '
+              \ . '$(stack --verbosity silent path --project-root)'
+              \ . '/*.cabal'
+        if has('nvim')
+            let job_id = jobstart(
+                  \   command
+                  \ , {'on_exit': function('s:CheckHdevPPAsync')}
+                  \ )
+            if job_id > 0
+                call jobwait([job_id])
```

Comment:
What is the benefit of using a job here, when waiting for it just below?
Callbacks for other jobs might happen, but otherwise it would still block, no?